src/components/Modal/index.tsx

```diff
@@ -0,0 +1,136 @@
+'use client'
+
+import * as React from 'react'
+import * as DialogPrimitive from '@radix-ui/react-dialog'
+import { CloseModal } from '@/assets'
+
+import { cn } from '@/lib/utils'
+
+const Modal = DialogPrimitive.Root
+
+const ModalTrigger = DialogPrimitive.Trigger
+
+const ModalPortal = DialogPrimitive.Portal
+
+const ModalCloseFooter = DialogPrimitive.Close
+
+const ModalOverlay = React.forwardRef<
+  React.ElementRef<typeof DialogPrimitive.Overlay>,
+  React.ComponentPropsWithoutRef<typeof DialogPrimitive.Overlay>
+>(({ className, ...props }, ref) => (
+  <DialogPrimitive.Overlay
+    ref={ref}
+    className={cn(
+      'fixed inset-0 z-50 bg-background/80 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0',
```

Comment:
Best to remove the animations I think. They might be elsewhere in this project, but that's usually a mistake.